### PR TITLE
Add: static enum parameter type.

### PR DESCRIPTION
**Pull request by rockwotj**
This adds simple enums from the discussion [here](https://discuss.redash.io/t/add-enum-support-for-query-parameters/366/5).

### CODE REVIEW

File: client/app/components/parameter-settings.html [hunk 1]
```diff
@@ -22,5 +23,9 @@ <h4 class="modal-title">{{$ctrl.parameter.name}}</h4>
         <label>Global</label>
         <input type="checkbox" class="form-inline" ng-model="$ctrl.parameter.global">
       </div>
+      <div class="form-group">
```
Comment (arikfr) [inline client/app/components/parameter-settings.html]: Shouldn't this show only if the user picked `enum` before?

File: client/app/components/parameter-settings.html [hunk 2]
```diff
@@ -13,6 +13,7 @@ <h4 class="modal-title">{{$ctrl.parameter.name}}</h4>
         <select ng-model="$ctrl.parameter.type" class="form-control">
           <option value="text">Text</option>
           <option value="number">Number</option>
+          <option value="enum">Enum</option>
```
Comment (arikfr) [inline client/app/components/parameter-settings.html]: I'm not a native English speaker, but will `Enum` be clear for non developers?

File: client/app/components/parameters.html [hunk 3]
```diff
@@ -6,6 +6,11 @@
       <input ng-switch-when="datetime-with-seconds" type="datetime-local" step="1" class="form-control" ng-model="param.ngModel">
       <input ng-switch-when="datetime-local" type="datetime-local" class="form-control" ng-model="param.ngModel">
       <input ng-switch-when="date" type="date" class="form-control" ng-model="param.ngModel">
+      <span ns-switch-when="enum">
+        <select ns-model="param.value" class="form-control">
+          <option ng-repeat="option in param.enumOptions.split('\n')" value="{{option}}">{{option}}</option>
```
Comment (arikfr) [inline client/app/components/parameters.html]: It's cleaner to do the split in the controller code (also gives more flexibility in the future).

File: client/app/components/parameter-settings.html (hunk 2 above)
Comment (rockwotj) [inline client/app/components/parameter-settings.html]: I don't think so, but this text is exposed to people writing queries only, will non-devs be writing queries often? I would think if you can write a query that you would know what an enum is. MySQL at least has a [Enum type](https://dev.mysql.com/doc/refman/5.7/en/enum.html). I'm fine to rename it if you'd like

File: client/app/components/parameters.html (hunk 3 above)
Comment (rockwotj) [inline client/app/components/parameters.html]: Is this what you intended?

File: client/app/components/parameter-settings.html (hunk 1 above)
Comment (rockwotj) [inline client/app/components/parameter-settings.html]: Oops.

File: client/app/components/parameter-settings.html (hunk 2 above)
Comment (arikfr) [inline client/app/components/parameter-settings.html]: Maybe not often, but they do and we should be friendly :-)

How about "Dropdown List"?

**Comment by arikfr** [comment]
I went ahead and renamed enum to dropdown and merged.

Thanks again for another great feature :)

**Comment by rockwotj** [comment]
Thanks for renaming and merging!

**Comment by arikfr** [comment]
Sure thing! Are you planning working on the dynamic one as well? 🙏

**Comment by rockwotj** [comment]
I have another feature I will add first having to do with the datepicker (we already use it internally), but after that I might, especially because updating the drop-down is annoying.

**Pull request merged**